boltz/btc.go

```diff
@@ -49,23 +50,53 @@ func (transaction *BtcTransaction) VSize() uint64 {
 }
 
 func (transaction *BtcTransaction) FindVout(network *Network, addressToFind string) (uint32, uint64, error) {
+	toFind, err := btcutil.DecodeAddress(addressToFind, network.Btc)
+	if err != nil {
+		return 0, 0, err
+	}
 	for vout, output := range transaction.MsgTx().TxOut {
-		_, outputAddresses, _, err := txscript.ExtractPkScriptAddrs(output.PkScript, network.Btc)
-
-		// Just ignore outputs we can't decode
-		if err != nil {
-			continue
-		}
-
-		for _, outputAddress := range outputAddresses {
-			if outputAddress.EncodeAddress() == addressToFind {
-				return uint32(vout), uint64(output.Value), nil
-			}
+		// first 2 bytes are witness type and length
+		if bytes.Equal(output.PkScript[2:], toFind.ScriptAddress()) {
```

Comment:
Add a check if the `output.PkScript` has more than 3 bytes. It's possible and just in case a tx like that is ever used with this function, it should not panic